components/atom/toast/src/index.js

```diff
@@ -0,0 +1,149 @@
+import React, {useState, useEffect, useRef, useCallback} from 'react'
+import PropTypes from 'prop-types'
+import IconClose from '@schibstedspain/sui-svgiconset/lib/Close'
+import cx from 'classnames'
+import {
+  AUTO_CLOSE_TIMES,
+  BASE_CLASS,
+  EFFECT_DELAY,
+  MARGINS,
+  POSITIONS,
+  SIZES
+} from './config'
+
+function AtomToast({
+  autoClose = true,
+  autoCloseTime = AUTO_CLOSE_TIMES.medium,
+  children,
+  crossToClose = true,
+  effect = true,
+  globalClose = false,
+  iconClose = <IconClose />,
+  margin = MARGINS.medium,
+  onClose = () => {},
+  position = POSITIONS.topRight,
+  show: showFromProps = true,
+  size = SIZES.medium
+}) {
+  const [show, setShow] = useState(showFromProps)
+  const [delay, setDelay] = useState(true)
+
+  const autoCloseTimeout = useRef()
+  const delayTimeout = useRef()
+  const toastRef = useRef()
+
+  const wrapperClassName = cx(
+    BASE_CLASS,
+    `${BASE_CLASS}-position--${position}`,
+    `${BASE_CLASS}-size--${size}`,
+    `${BASE_CLASS}-margin--${margin}`,
+    {
+      [`${BASE_CLASS}-effect--${position}`]: effect,
+      [`${BASE_CLASS}-effect--hide`]: effect && delay
+    }
+  )
+
+  const handleClose = useCallback(() => {
+    if (effect) setDelay(true)
+    setShow(false)
+    if (!effect) onClose()
+  }, [effect, onClose])
+
+  useEffect(() => {
+    setShow(showFromProps)
+  }, [showFromProps])
+
+  useEffect(() => {
+    if (autoClose) {
+      autoCloseTimeout.current = setTimeout(() => {
+        handleClose()
+      }, autoCloseTime)
+    }
+
+    return () => clearTimeout(autoCloseTimeout.current)
+  }, [autoClose, autoCloseTime, handleClose])
+
+  useEffect(() => {
+    // open effect
+    if (effect && show && delay) {
+      delayTimeout.current = setTimeout(() => {
+        setDelay(false)
+      }, EFFECT_DELAY.open)
+    }
+
+    // close effect
+    if (effect && !show) {
+      delayTimeout.current = setTimeout(() => {
+        setDelay(false)
+        onClose()
+      }, EFFECT_DELAY.close)
+    }
+
+    return () => clearTimeout(delayTimeout.current)
+  }, [delay, effect, onClose, show])
+
+  useEffect(() => {
+    if (globalClose) {
+      const handleClickOutside = e => {
+        if (!toastRef.current.contains(e.target)) {
+          handleClose()
+        }
+      }
+      window.addEventListener('mousedown', handleClickOutside)
```

Comment:
I think this could be improved by avoiding adding an event to the entire window but I don't know how

Reply:
This looks like a possible hook `useOutsideClick`